# HG changeset patch
# User jpswensen@compsci34-82-1651.compscidhcp.jhu.edu
# Date 1224789771 14400
#      Thu Oct 23 15:22:51 2008 -0400
# Node ID ed5811a1ec8f9a0c5dd094db5b666cd87e8a1dc7
# Parent  095b3e4d64e99065bba14a14fcf849cc68a14fff
Added output about why a library load fails on OSX

diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -1,8 +1,13 @@
+2008-10-23  John Swensen  <jpswensen@comcast.net>
+
+	* oct-shlib.cc (octave_dyld_shlib::open): Call NSLinkEditError to
+	get better diagnostic if NSLinkModule fails.
+
 2008-10-23  John W. Eaton  <jwe@octave.org>
 
 	* oct-shlib.cc (octave_dlopen_shlib::open): Use RTLD_NOW instead
 	of RTLD_LAZY.
 
 2008-10-12  Jaroslav Hajek <highegg@gmail.com>
 
 	* CSparse.cc (ComplexMatrix::expm): Improve inverse preconditioning
diff --git a/liboctave/oct-shlib.cc b/liboctave/oct-shlib.cc
--- a/liboctave/oct-shlib.cc
+++ b/liboctave/oct-shlib.cc
@@ -606,18 +606,26 @@ octave_dyld_shlib::open (const std::stri
 				  | NSLINKMODULE_OPTION_PRIVATE));
 	  if (handle)
 	    {
 	      stamp_time ();
 	      isOpen = true;
 	    }
 	  else
 	    {
-	      (*current_liboctave_error_handler)
-		("couldn't link module %s", file.c_str ());	
+	      NSLinkEditErrors ler;
+	      int lerno;
+	      const char *file2;
+	      const char *errstr = 0;
+
+	      NSLinkEditError (&ler, &lerno, &file2, &errstr);
+
+	      if (errstr)
+		(*current_liboctave_error_handler)
+		  ("%s: %s", file.c_str (), errstr);	
 	    }
 	}
       else
 	{
 	  (*current_liboctave_error_handler)
 	    ("got NSObjectFileImageReturnCode %d", returnCode);
 
 	  // FIXME -- should use NSLinkEditError () to get
